pkg/profiles/profiles.go

```diff
@@ -380,6 +370,18 @@ func Validate(data []byte) error {
 		}
 	}
 
+	rb := serviceProfile.Spec.RetryBudget
+	if rb != nil {
+		if rb.MinRetriesPerSecond == 0 || rb.RetryRatio == 0 || rb.TTL == "" {
```

Comment:
I think we need to allow zero values here.  (don't get me started about how mad I am that you can't differentiate between 0 and unset 😡)

A more appropriate validation would be that `minRetriesPerSecond` and `retryRatio` cannot be negative.

Reply:
@adleong My assumption was that because these are now required fields, they should not be set to zero?

Related, the `if rb != nil {` check above this should handle the case where no `RetryBudget` is set.

Reply:
Another related question: is it possible to make these fields `*int`, etc, to indicate whether it was set?

Reply:
It must be possible to explicitly set them to 0.  i.e.

```
retryBudget:
  retryRatio: 0.0
  minRetriesPerSecond: 0
  ttl: 10m
```

Reply:
> Another related question: is it possible to make these fields `*int`, etc, to indicate whether it was set?

I don't know.  Is this an existing pattern?  I will investigate but in the meantime, I think this validation needs to allow 0 values.

Reply:
Agreed that is not a change I'd want to make in this PR, but it is an existing pattern:
https://github.com/linkerd/linkerd2/blob/a5628780f7dbeff47549c7270af5549ec53c0434/cli/cmd/routes.go#L229-L242